### ISSUE TITLE
:echo nvim_get_all_options_info() crashes Neovim

### ISSUE THREAD

**Issue by dstein64** [neovim/neovim#13962]
- `nvim --version`: NVIM v0.5.0-dev+1093-g4bc74c243
- `vim -u DEFAULTS` (version: ) behaves differently? N/A
- Operating system/version: Ubuntu 20.04.2
- Terminal name/version: Konsole 19.12.3
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:echo nvim_get_all_options_info()
```

### Actual behaviour

Neovim crashes with the following output:

```
nvim: ../src/nvim/eval/encode.c:840: encode_tv2echo: Assertion `eve_ret == OK' failed.
                                                                                      Aborted
```

### Expected behaviour

Option information would be printed.

### Additional information

This option is documented in `$VIMRUNTIME/doc/api.txt` (`:help nvim_get_all_options_info()`).

Using `call` instead of `echo` results in an error, without crashing.

```
:call nvim_get_all_options_info()
```
```
Error converting the call result: Empty dictionary keys aren't allowed
```

**Comment by bfredl** [comment]
Fixed in #13479 IIRC

**Comment by glacambre** [comment]
Fixed in https://github.com/neovim/neovim/pull/14607 I think.
